lib/kitchen/driver/rackspace.rb

```diff
@@ -168,19 +184,34 @@ def servicelevel_check(server)
       end
 
       def hostname(server)
-        if config[:servicenet] == false
+        if config[:ssh_network_name]
+          server.addresses[config[:ssh_network_name]].first['addr']
+        elsif config[:connect_public_net] && config[:servicenet] == false
           server.public_ip_address
         else
           server.private_ip_address
         end
       end
 
+      private def public_network
+        # Returns nil if no public ip address should be provisioned
+        '00000000-0000-0000-0000-000000000000' if config[:connect_public_net]
+      end
+
+      private def service_network
```

Comment:
I thought this was known as the "private" network?  Is "service" a synonym in Rackspace terms?